### PR TITLE
Added assertion tasks to ensure user is logged into the right cluster

### LINKED ISSUE #73: Add assertion for Kubeconfig

### PR DESCRIPTION
**Pull request by pranavgaikwad**
Fixes #73

### CODE REVIEW

File: common/assert-kubeconfig.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        (k8s_version is defined and
+        web_console_url is defined)
+  debug:  
+    msg: |
+      {% if openshift_version is defined and openshift_version.major is defined %}
+      Detected OpenShift version - {{ openshift_version.major|string + '.' + openshift_version.minor|string }}
+      {% elif openshift_version is defined and openshift_version.major is not defined %}
+      Detected OpenShift version - {{ openshift_version|string }}
+      {% elif k8s_version is defined %}
+      Detected k8s version - {{ k8s_version.major|string + '.' + k8s_version.minor|string }}
```
Comment (alaypatel07) [inline common/assert-kubeconfig.yml]: would it be too much work to map k8s minor version to openshift 3 version here? 

```
1.7 -> 3.7
1.8 -> 3.8
1.9 -> 3.9
```

would be nice to have for folks who are unaware of the mapping

**Comment by alaypatel07** [comment]
before we merge this please also have a cherry-pick PR onto stunnel branch. Thanks

**Comment by pranavgaikwad** [comment]
@alaypatel07 Opened new PR against `stunnel` branch : https://github.com/konveyor/pvc-migrate/pull/97

**Review by alaypatel07 — APPROVED** [review]
@pranavgaikwad verified we dont need k8s->openshift version mapping. This looks good, thanks

File: common/assert-kubeconfig.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  - set_fact:
+      openshift_version_info: "{{ lookup('k8s', 
+        api_version='config.openshift.io/v1', 
+        kind='ClusterVersion',
+        resource_name='version'
+      ) }}"
+  
+  - set_fact:
+      openshift_version: "{{ openshift_version_info.status.desired.version }}"
+    when: openshift_version_info is defined and
```
Comment (djwhatle) [inline common/assert-kubeconfig.yml]: You might be interested to know that this `when` check can be performed in on one line:
```
when: openshift_version_info.get("status", {}).get("desired", {}).get("version, "") != ""
```

Syntax above may not be 100% correct, but general idea is that you can do a chain of calls like:
```
# you can run python "get" on ansible dicts. I use an empty dict as the fallback value
# except on the final get call where we will compare against an empty string
get("desiredKey", fallBackIfNotFound) 
```


The end result here is the same, so feel free to leave as-is.

**Comment by djwhatle** [comment]
Quickly trying this out, but looks good 👍 .

**Comment by djwhatle** [comment]
@pranavgaikwad I updated the warning message to match Stage 1 message. Warning worked for me, think this will save folks from mistakes 👍 

image

**Review by djwhatle — APPROVED** [review]

**Review by pranavgaikwad — COMMENTED** [review]
LGTM

**Pull request merged**